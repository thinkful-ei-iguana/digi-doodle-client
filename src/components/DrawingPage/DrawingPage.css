.player-li {
    list-style-type: none;
    margin-top: 1.5%;
    font-size: 24px;
  }
  .player-header {
    font-size: 26px;
  }
  .player-prompt {
    font-size: 22px;
  }
  body {
    background-image: url("../../Pictures/paper-background.jpg");
    background-position: center;
    background-position-x: 25%;
  }
  @media screen and (min-width: 900px) {
      .color-buttons {
          margin-top: 1%;
      }
  }