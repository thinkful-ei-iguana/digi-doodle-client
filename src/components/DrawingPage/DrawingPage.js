import React, { Component } from 'react'
import Canvas from '../../Utils/Canvas/Canvas'
import Colors from '../../Utils/Colors/Colors'
import ColorContext from '../../Context/ColorContext'
import socket from '../../services/socket-service'
import PlayAgain from '../PlayAgain/PlayAgain'
import '../../Utils/Colors/Colors.css'
import './DrawingPage.css'
import '../../Utils/Canvas/Canvas.css'

export default class DrawingPage extends Component {
    static contextType = ColorContext
    constructor(props) {
        super(props);
        this.state = {
            guess: '',
            username: '',
            players: [],
            score: 0
        }
    }

    static contextType = ColorContext


    handleChatSubmit = async (ev) => {
        ev.preventDefault();

        socket.emit('chat message', { player: this.context.username, message: this.state.guess });

        // console.log('guess response: ', guess);
        await this.setState({
            guess: ''
        });
    }

    handleTextInput = (ev) => {
        this.setState({
            guess: ev.target.value
        })
    }


    render() {
        let disableAttr = this.props.isDrawing ? "canvas-container" : "disabled-canvas"
        let standbyMode = false;
        if (this.context.game && this.context.game.status) {
            standbyMode = this.context.game.status === 'standby'
        }
        disableAttr = disableAttr || standbyMode

        let toggleDisplay;

        if (this.context.game && this.context.game.winner) {
            toggleDisplay = <PlayAgain />
        } else {
            toggleDisplay =
                <div className="chat-window">
                    <ul>
                        {this.context.messages && this.context.messages.map((message, index) => {
                            return (
                                <li className="player-message" key={index}>{message.player}: {message.message}</li>
                            )
                        })}
                    </ul>
                </div>
        }

        return (
            <div>
                <div className={disableAttr}>
                    <Canvas />
                </div>
                <Colors />
                <div className="players-container">
                    <ul className="player-ul">
                        {this.context.players && this.context.players.map((player, index) => {
                            return (
                                <li className="player-li" key={index}>
                                    <span>{player.username}</span><br />
                                    <span className="score">{player.score}</span>
                                </li>
                            )
                        })}
                    </ul>
                </div>

                {toggleDisplay}

                <p className="reminder-instructions">15 points to win</p>
            </div>
        )
    }
}
