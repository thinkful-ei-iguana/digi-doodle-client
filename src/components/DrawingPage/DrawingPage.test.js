import React from 'react'
import ReactDOM from 'react-dom'
import DrawingPage from './DrawingPage'
import Canvas from '../../Utils/Canvas/Canvas'
import { mount, shallow, render } from 'enzyme'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

describe(`Drawing component`, () => {
    it('renders without crashing', () => {

        const context = {
            game: {
                id: 'game uuid',
                current_drawer: 'test player',
                current_answer: 'test',
                status: 'waiting for players',
                winner: 'test winner'
            }
        }

        const drawingPage = shallow(<DrawingPage />, { context })
        drawingPage.setContext({ context })
        expect(drawingPage).toMatchSnapshot();
    })
})