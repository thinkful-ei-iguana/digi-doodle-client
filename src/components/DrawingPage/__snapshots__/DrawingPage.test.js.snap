// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawing component renders without crashing 1`] = `ShallowWrapper {}`;
